version: "3.2"

services:
    agent:
        image: portainer/agent:lts
        volumes:
            - /var/run/docker.sock:/var/run/docker.sock
            - /var/lib/docker/volumes:/var/lib/docker/volumes
        networks:
            - agent_network
        deploy:
            mode: global
            placement:
                constraints: [node.platform.os == linux]

    portainer:
        image: portainer/portainer-ce:lts
        command: -H tcp://tasks.agent:9001 --tlsskipverify
        ports:
            - "9443:9443"
            - "9000:9000"
            - "8000:8000"
        volumes:
            - portainer_data:/data
            - "/var/run/docker.sock:/var/run/docker.sock"
        networks:
            - agent_network
            - app_network
        deploy:
            mode: replicated
            replicas: 1
            placement:
                constraints: [node.role == manager]
            restart_policy:
                condition: on-failure
            labels:
                - "traefik.enable=true"
                - "traefik.http.routers.portainer.rule=Host(`portainer.booboofashions.com`)"
                - "traefik.http.routers.portainer.entrypoints=websecure"
                - "traefik.http.routers.portainer.tls.certresolver=myresolver"
                - "traefik.http.services.portainer.loadbalancer.server.port=9000"
                - "traefik.docker.network=app_network"

networks:
    agent_network:
        external: true
    app_network:
        external: true

volumes:
    portainer_data:
        external: true
